zingolib/src/lightclient.rs

```diff
@@ -946,16 +947,58 @@ impl LightClient {
             "outgoing_metadata" => outgoing_json,
         }
     }
-    /*
-        pub async fn do_list_txsummaries(&self) -> JsonValue {
-            for (txid, transaction_md) in self.wallet.transaction_context.transaction_metadata_set.read().await.current.iter()
-            {
-                let tx_value = transaction_md.total_value_spent();
-                let to_addresses =
-    }
-            Vec<TxSummary>
+    pub async fn do_list_txsummaries(&self) -> Vec<ValueTransferSummary> {
+        let mut sends = Vec::new();
+        let mut receipts = Vec::new();
+
+        for (txid, transaction_md) in self
+            .wallet
+            .transaction_context
+            .transaction_metadata_set
+            .read()
+            .await
+            .current
+            .iter()
+        {
+            let tx_value_spent = transaction_md.total_value_spent();
```

Comment:
Shall we put a comment here saying that this includes tx_fees?   Or is there another more logical place.   I guess a place that `rustc` reads is best.